docs/docs/spec.md

```diff
@@ -351,7 +351,8 @@ innermost.
 > defining that there exists a series of enclosing declarations for any type
 > usage, not just for a declaration.
 
-At each declaration, check the following rules in order:
+At each declaration that is a [recognized](#recognized-declaration) location,
```

Comment:
```suggestion
At each declaration that is in a [recognized](#recognized-declaration) location,
```

Reply:
Hmm, I think of the declaration itself as _being_ the recognized location: A method is a recognized location for a `@Target(METHOD)` annotation, etc.

I'll merge this PR as it stands now, but we can continue discussing if you'd like.